File: javaDemo/src/main/java/globus/demo/MapTextureViewActivity.java
package globus.demo;

import android.app.Activity;
import android.os.Bundle;
import android.view.TextureView;

import globus.glmap.GLMapStyleParser;
import globus.glmap.GLMapVectorCascadeStyle;
import globus.glmap.GLMapView;
import globus.glmap.MapPoint;

public class MapTextureViewActivity extends Activity {
  GLMapView mapView;

  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.texture_view_map);

    TextureView textureView = findViewById(R.id.texture_view);
    mapView = new GLMapView(this, textureView);

    GLMapVectorCascadeStyle style = new GLMapStyleParser(getAssets(), "DefaultStyle.bundle").parseFromResources();
    if (style != null)
      mapView.setStyle(style);
    mapView.setScaleRulerStyle(GLMapView.GLMapPlacement.BottomCenter, 10, 10, 200);
    mapView.setAttributionPosition(GLMapView.GLMapPlacement.TopCenter);
  }
}

File: kotlinDemo/src/main/java/globus/demo/MapTextureViewActivity.kt
package globus.demo

import android.app.Activity
import android.os.Bundle
import android.view.TextureView
import globus.glmap.GLMapStyleParser
import globus.glmap.GLMapView
import globus.glmap.GLMapTextureView
import globus.glmap.GLMapViewRenderer
import globus.glmap.MapPoint

class MapTextureViewActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.texture_view_map)
        val textureView = findViewById<GLMapTextureView>(R.id.texture_view)
        textureView.renderer.setStyle(GLMapStyleParser(assets, "DefaultStyle.bundle").parseFromResources()!!)
        textureView.renderer.setScaleRulerStyle(GLMapViewRenderer.GLMapPlacement.BottomCenter, 10, 10, 200.0)
        textureView.renderer.setAttributionPosition(GLMapViewRenderer.GLMapPlacement.TopCenter)
    }
}
